### PR TITLE
Refactor runReservePluginUnreserve

### PR DESCRIPTION
**Pull request by yuzhiquan**
**What type of PR is this?**

 /kind cleanup

**What this PR does / why we need it**:
refactor runReservePluginUnreserve fuction
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/runtime/framework.go [hunk 1]
```diff
@@ -829,13 +829,11 @@ func (f *frameworkImpl) RunReservePluginsUnreserve(ctx context.Context, state *f
...
-	if !state.ShouldRecordPluginMetrics() {
-		pl.Unreserve(ctx, state, pod, nodeName)
-		return
-	}
 	startTime := time.Now()
 	pl.Unreserve(ctx, state, pod, nodeName)
-	f.metricsRecorder.observePluginDurationAsync(unreserve, pl.Name(), nil, metrics.SinceInSeconds(startTime))
+	if state.ShouldRecordPluginMetrics() {
+		f.metricsRecorder.observePluginDurationAsync(unreserve, pl.Name(), nil, metrics.SinceInSeconds(startTime))
+	}
```
Comment (chendave) [inline pkg/scheduler/framework/runtime/framework.go:836]: only tiny difference, 
- there is a extra check statement while metrics should be recorded, not sure how this impacts the metrics data, ignorable?
- `startTime` become unnecessary when the metrics should not be recorded.

Comment (yuzhiquan) [reply pkg/scheduler/framework/runtime/framework.go:836]: - `ShouldRecordPluginMetrics` depend on the variable `recordPluginMetrics`, looks like not last too long.
-  Yes.

**Comment by alculquicondor** [comment]
/unassign
/assign @adtac 

why is it important for this to be in 1.19?

**Comment by yuzhiquan** [comment]
> /unassign
> /assign @adtac
> 
> why is it important for this to be in 1.19?

Noop, that's just an refactor, not Imperative for 1.19.

**Comment by adtac** [comment]
I'm not sure what this refactor adds (is there a precedent of this pattern in other places?), but I'm mostly indifferent to the change. However, I'll mention that I noticed time.Now() consuming a non-negligible chunk of the scheduler's time when I profiled the scheduler, so I'd be wary of introducing unnecessary calls to time.Now().

**Pull request closed without merging**